src/NHibernate/Transform/CacheableResultTransformer.cs

```diff
@@ -225,12 +232,14 @@ public object TransformTuple(object[] tuple, string[] aliases)
 		/// <param name="aliases">The aliases that correspond to the untransformed tuple.</param>
 		/// <param name="transformer">The transformer for the re-transformation.</param>
 		/// <param name="includeInTuple"></param>
-		/// <returns>transformedResults, with each element re-transformed (if necessary).</returns>
+		/// <returns><paramref name="transformedResults"/>, with each element re-transformed (if necessary).</returns>
 		public IList RetransformResults(IList transformedResults,
 		                                string[] aliases,
 		                                IResultTransformer transformer,
 		                                bool[] includeInTuple)
 		{
+			if (transformedResults.Count == 0)
+				return transformedResults;
```

Comment:
This short-circuit, and the one in `UntransformToTuples`, could be removed for a single one in the `Loader`, which is the only caller of these methods. But it seems more appropriate to me to put them here, especially since their code documentation states:

> If not necessary, the original List is returned unchanged.